src/filter_tree/filter_tree.c

```diff
@@ -132,12 +132,7 @@ Vector *FilterTree_SubTrees(const FT_FilterNode *root) {
 /* Applies a single filter to a single result.
  * Compares given values, tests if values maintain desired relation (op) */
 int _applyFilter(SIValue *aVal, SIValue *bVal, AST_Operator op) {
-	int rel = SIValue_Compare(*aVal, *bVal);
-	/* Values are of disjoint types */
-	if(rel == DISJOINT) {
-		/* The filter passes if we're testing for inequality, and fails otherwise. */
-		return (op == OP_NEQUAL);
-	}
+	int rel = SIValue_Order(*aVal, *bVal);
```

Comment:
Logically, filters should test for comparability and not orderability. The basic distinction is that when 2 values can't be compared because of their types, they can still be ordered based on Cypher's established sort order of types, but it doesn't make sense to have a construction like `MATCH (a) WHERE a < a.numeric_prop`.

Reply:
very good comment! I was under the impression that `return 1 < "1" ` is a valid expression